### PR TITLE
ci: run airbyte connector tests from master branch

### PR DESCRIPTION
**Pull request by aaronsteers**
This reverts a previous change that pulled airbyte-ci and connector images from a fixed branch where we were iterating on airbyte-ci improvements.

After this change:
- All connector definitions in the "Test Connectors" workflow will be pulled from `master` branch on the monorepo.
- Manifest-only connectors (PokeAPI and The Guardian) are removed from the matrix since they otherwise would represent a false positive success. The work to enable them again is logged here:
  - https://github.com/airbytehq/airbyte-python-cdk/issues/64

## Summary by CodeRabbit

- **Chores**
	- Updated workflow configuration for testing connectors to enhance clarity and functionality.
	- Improved comments regarding filters and job cancellation logic.
	- Refined connector references for better readability without altering functionality.

### CODE REVIEW

File: .github/workflows/connector-tests.yml [hunk 1]
```diff
@@ -74,21 +74,18 @@ jobs:
             cdk_extra: n/a
           - connector: source-chargebee
             cdk_extra: n/a
-          # Currently not passing CI (unrelated)
```
Comment (maxi297) [inline .github/workflows/connector-tests.yml:77]: This should be fixed now. Why are we removing source-zendesk-support?

**Review by maxi297 — COMMENTED** [review]
LGTM! One question regarding zendesk-support

File: .github/workflows/connector-tests.yml (hunk 1 above)
Comment (aaronsteers) [reply .github/workflows/connector-tests.yml:77]: @maxi297 - Happy to restore it. I'll do that now.

Comment (aaronsteers) [reply .github/workflows/connector-tests.yml:77]: Done ✅

Comment (maxi297) [reply .github/workflows/connector-tests.yml:77]: It seems like `ticket_activities` is missing some records. I don't know if they expire. @DanyloGL would you be able to check this one?

File: .github/workflows/connector-tests.yml [hunk 2]
```diff
@@ -116,8 +117,7 @@ jobs:
         if: steps.no_changes.outputs.status != 'cancelled'
         with:
           repository: airbytehq/airbyte
-          # TODO: Revert to `master` after Airbyte CI released:
-          ref: aj/airbyte-ci/update-python-local-cdk-code
+          ref: master
```
Comment (aldogonzalez8) [inline .github/workflows/connector-tests.yml:120]: 👍

**Pull request merged**